source/agora/consensus/validation/Block.d

```diff
@@ -136,7 +140,34 @@ public string isInvalidReason (const ref Block block, Height prev_height,
             return fail_reason;
     }
 
-    return null;
+    return validateBlockTimestamp(prev_timestamp, block.header.timestamp, curr_timestamp, blockTimestampToleranceSecs);
+}
+
+static public string validateBlockTimestamp(ulong prev_block_ts, ulong new_block_ts, ulong curr_timestamp,
```

Comment:
`static` has no meaning in module scope.

Also this function is public and doesn't have documentation.